src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,109 @@
+package duke;
+
+import java.util.ArrayList;
+
+public class TaskList {
+    private ArrayList<Task> tasks;
+    
+    /**
+     * Constructor for TaskList
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public int getSize() {
+        return tasks.size();
+    }
+
+    public Task getTask(int index) {
+        return tasks.get(index);
+    }
+
+    /**
+     * summarize prints out all tasks inside a TaskList object.
+     * @param
+     * @return String with all tasks
+     */
+    public String summarize() {
+        String all_tasks = "Here are the tasks in your list:\n";
+        for (int i = 0; i < this.tasks.size(); i++) {
+            Task t = tasks.get(i);
+            all_tasks += String.format("%d.%s\n", i+1, t.toString());
+        }
+        return all_tasks;
+    }
+
+    /**
+     * mark_done helps change the status of any Task inside TaskList, and mark
```

Comment:
The additional noun here (mark_done) could be removed ☺️ Just verbs are enough if we are following our cs2103 coding style 🤓